ui/analyse/src/ctrl.ts

```diff
@@ -378,6 +378,9 @@ export default class AnalyseCtrl {
     this.setAutoShapes();
     if (this.node.shapes) this.chessground.setShapes(this.node.shapes as DrawShape[]);
     this.cgVersion.dom = this.cgVersion.js;
+    site.pubsub.on('theme.change', () => {
+      this.chessground.state.addPieceZIndex = $('#main-wrap').hasClass('is3d');
```

Comment:
I tried, this doesn't work. Chessground doesn't redraw its piece when this attribute is changed.

Reply:
it works for newly moved pieces though, it was enough in my testing.

Reply:
when transitioning from 2d to 3d, the entire board is at z-index 2 but is drawn correctly. So it really depends on whether you move something behind a tall piece that hasn't been moved since we went 3d.

Why not just always add z-index for big boards (on 2d too)?

Reply:
Ah, I did not try moving a piece behind a tall piece that had not been moved yet since the transition.